File: frontend/wms-web/src/types/purchaseOrder.ts
export type PurchaseOrderStatus =
  | 'Draft'
  | 'Approved'
  | 'PartiallyReceived'
  | 'Completed'
  | 'Cancelled'

export type PurchaseOrderLine = {
  id: number
  productId: number
  productSku: string
  productName: string
  orderedQuantity: number
  receivedQuantity: number
  remainingQuantity: number
}

export type PurchaseOrder = {
  id: number
  orderNumber: string
  warehouseId: number
  warehouseName: string
  supplierName: string
  status: PurchaseOrderStatus
  isActive: boolean
  createdAtUtc: string
  receivedAtUtc: string | null
  completedAtUtc: string | null
  lines: PurchaseOrderLine[]
}

export type PurchaseOrderLineRequest = {
  productId: number
  quantity: number
}

export type SavePurchaseOrderRequest = {
  warehouseId: number
  supplierName: string
  lines: PurchaseOrderLineRequest[]
}

File: backend/Wms.Api/Domain/Entities/PurchaseOrder.cs
using Wms.Api.Domain.Enums;

namespace Wms.Api.Domain.Entities;

public class PurchaseOrder
{
    public int Id { get; set; }
    public required string OrderNumber { get; set; }
    public int WarehouseId { get; set; }
    public required string SupplierName { get; set; }
    public DateTime? ReceivedAtUtc { get; set; }
    public bool IsActive { get; set; } = true;
    public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;
    public PurchaseOrderStatus Status { get; set; } = PurchaseOrderStatus.Draft;
    public Warehouse Warehouse { get; set; } = null!;
    public ICollection<PurchaseOrderLine> Lines { get; set; } = [];
}
